src/Bicep.Core/Parser/Parser.cs

```diff
@@ -484,6 +485,19 @@ private IdentifierSyntax Identifier(DiagnosticBuilder.ErrorBuilderDelegate error
             return new IdentifierSyntax(identifier);
         }
 
+        private IdentifierSyntax IdentifierOrSkip(DiagnosticBuilder.ErrorBuilderDelegate errorFunc)
```

Comment:
>IdentifierOrSkip [](start = 33, length = 16)

This ended up being much less greedy than our normal recovery logic. Skipping to the end of the line would consume too many token and cause some weird behaviors.